python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -125,7 +125,7 @@ def notify_resource_usage(
         free_slots = self._actor_pool.num_free_slots()
         if input_queue_size > free_slots and under_resource_limits:
             # Try to scale up if work remains in the work queue.
-            self._scale_up_if_needed()
+            self._scale_up_if_needed(num_inputs=input_queue_size)
```

Comment:
It seems we do the `input_queue_size > free_slots` check twice: first in the if block above, and second in scale_up_if_needed. Is it possible to eliminate one of these checks?

Reply:
updated